### ISSUE TITLE
Create `ArtifactRequestHandler` for IDS Multipart

### ISSUE THREAD

**Issue by DominikPinsel** [eclipse-dataspaceconnector/DataSpaceConnector#231]
Hello everyone,

for this issue an `ArtifactRequestHandler` should be added to the IDS Multipart Extension. 

- the new `ArtifactRequestHandler` must be able to handle an IDS `ArtifactRequestMessage` ([link](https://github.com/International-Data-Spaces-Association/Java-Representation-of-IDS-Information-Model/blob/main/de/fraunhofer/iais/eis/ArtifactRequestMessage.java))
- before the asset is transferred the handler should assert that the requestor has a valid DAT and `Contract`
- if the request valid, the `ArtifactRequestHandler` should 
  - return a `RequestInProcessMessage` ([link](https://github.com/International-Data-Spaces-Association/Java-Representation-of-IDS-Information-Model/blob/main/de/fraunhofer/iais/eis/RequestInProcessMessage.java))
  -  register a new `TransferProcess` in the `TransferProcessStore`([link](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/spi/src/main/java/org/eclipse/dataspaceconnector/spi/transfer/store/TransferProcessStore.java)) 
- for invalid requests return a `RejectionMessage`
- the sending of the `ArtifactResponseMessage` ([link](https://github.com/International-Data-Spaces-Association/Java-Representation-of-IDS-Information-Model/blob/main/de/fraunhofer/iais/eis/ArtifactResponseMessage.java)) is done by (or after) the transfer process and out of scope of this issue
- an incomplete example implementation was done in the `ArtifactRequestController` and can be used as reference ([link](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/data-protocols/ids/ids-api-transfer/src/main/java/org/eclipse/dataspaceconnector/ids/api/transfer/ArtifactRequestController.java))

> Depends on #230  

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

**Comment by jimmarino** [comment]
A small note, as part of this PR it would also be good to delete the `ArtifactRequestController` and any of the other related classes that are no longer used.

**Comment by DominikPinsel** [comment]
FYI: I will start working on this story

**Comment by DominikPinsel** [comment]
Done in PR
